shell/platform/windows/system_utils_unittests.cc

```diff
@@ -7,71 +7,46 @@
 
 #include "flutter/fml/macros.h"
 #include "flutter/shell/platform/windows/system_utils.h"
+#include "flutter/shell/platform/windows/testing/mock_windows_proc_table.h"
 #include "gtest/gtest.h"
 
 namespace flutter {
 namespace testing {
 
-class MockWindowsRegistry : public WindowsRegistry {
- public:
-  MockWindowsRegistry() = default;
-  virtual ~MockWindowsRegistry() = default;
-
-  virtual LSTATUS GetRegistryValue(HKEY hkey,
-                                   LPCWSTR key,
-                                   LPCWSTR value,
-                                   DWORD flags,
-                                   LPDWORD type,
-                                   PVOID data,
-                                   LPDWORD data_size) const {
-    using namespace std::string_literals;
-    static const std::wstring locales =
-        L"en-US\0zh-Hans-CN\0ja\0zh-Hant-TW\0he\0\0"s;
-    static DWORD locales_len = locales.size() * sizeof(wchar_t);
-    if (data != nullptr) {
-      if (*data_size < locales_len) {
-        return ERROR_MORE_DATA;
-      }
-      std::memcpy(data, locales.data(), locales_len);
-      *data_size = locales_len;
-    } else if (data_size != NULL) {
-      *data_size = locales_len;
-    }
-    return ERROR_SUCCESS;
-  }
-
- private:
-  FML_DISALLOW_COPY_AND_ASSIGN(MockWindowsRegistry);
-};
-
 TEST(SystemUtils, GetPreferredLanguageInfo) {
-  WindowsRegistry registry;
-  std::vector<LanguageInfo> languages = GetPreferredLanguageInfo(registry);
+  WindowsProcTable proc_table;
+  std::vector<LanguageInfo> languages =
+      GetPreferredLanguageInfo(WindowsProcTable());
   // There should be at least one language.
   ASSERT_GE(languages.size(), 1);
   // The info should have a valid languge.
   EXPECT_GE(languages[0].language.size(), 2);
 }
 
 TEST(SystemUtils, GetPreferredLanguages) {
-  WindowsRegistry registry;
-  std::vector<std::wstring> languages = GetPreferredLanguages(registry);
+  MockWindowsProcTable proc_table;
+  ON_CALL(proc_table, GetThreadPreferredUILanguages)
+      .WillByDefault(
+          [](DWORD flags, PULONG count, PZZWSTR languages, PULONG size) {
+            static const wchar_t lang[] = L"en-US\0\0";
+            static const size_t lang_len = sizeof(lang) / sizeof(wchar_t);
```

Comment:
Is `sizeof(lang)` right here? I'd expect this to be `wcslen`

Reply:
`lang` is a `wchar_t[...]`, so `sizeof(lang) / sizeof(wchar_t)` returns the number of characters in `lang` including the terminating null. I've confirmed that, when the extra null I erroneously added is removed (which I will push soon), the expression evaluates to 7, which is correct.

Reply:
Huh, TIL, I would've expected that to return the size of a pointer. Thanks for the details!